tracing-forest/src/layer/mod.rs

```diff
@@ -280,6 +280,71 @@ where
     }
 }
 
+/// A [`Layer`] that passes changes to span fields through as events.
+///
+/// Use this layer if you'd like notifications of when spans change their fields.
+///
+/// # Examples
+/// ```
+/// use tracing::{trace_span};
+/// use tracing_forest::{ForestLayer, SpanFieldEventLayer, traits::*};
+/// use tracing_subscriber::Registry;
+///
+/// Registry::default()
+///     .with(SpanFieldEventLayer::default())
+///     .with(ForestLayer::default())
+///     .init();
+///
+/// // -- snip --
+///
+/// let span = trace_span!("span_with_parameter", parameter = "See the trees");
+/// let _enter = span.enter();
+/// span.record("parameter", "for the forest");
+/// ```
+/// Produces the the output:
+/// ```log
+/// TRACE    ┕━ span_with_parameter [ 390µs | 2.66% ] parameter: "See the trees"
+/// TRACE       ┕━ 📍 [trace]:  | parameter: "for the forest"
+/// ```
+#[derive(Default)]
+pub struct SpanFieldEventLayer {}
+
+impl<S> Layer<S> for SpanFieldEventLayer
+where
+    S: Subscriber + for<'a> LookupSpan<'a>,
+{
+    fn on_record(&self, span: &Id, values: &Record<'_>, ctx: Context<'_, S>) {
+        let current_span = ctx.span(span).expect(fail::SPAN_NOT_IN_CONTEXT);
+        let mut fields = FieldSet::new();
+
+        values.record(&mut |field: &Field, value: &dyn fmt::Debug| {
+            let value = format!("{:?}", value);
+            fields.push(tree::Field::new(field.name(), value));
+        });
+
+        let shared = tree::Shared {
+            #[cfg(feature = "uuid")]
+            uuid: Uuid::nil(),
+            #[cfg(feature = "chrono")]
+            timestamp: Utc::now(),
+            level: *current_span.metadata().level(),
```

Comment:
nit: it might be useful if this were configurable.

What if `SpanFieldEventLayer` had a `level: Option<Level>` field where, if present, created events at that level and if absent, events inherited the level of their span (current solution)?

Reply:
Yes. I like that. I thought about having a value set in the layer, but thought the span level was more appropriate.

One thing that is a bit funky (with all of this, I suppose) is that the initial field values are logged on the line of the span and thus have the level of the span but any updates might have a different level. However, I don't think that's too much of an issue.